Source/WebCore/platform/graphics/skia/GraphicsContextSkia.h

```diff
@@ -27,25 +27,36 @@
 
 #if USE(SKIA)
 
+#include "GLFence.h"
 #include "GraphicsContext.h"
 WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN
 #include <skia/core/SkCanvas.h>
+#include <skia/core/SkImage.h>
 #include <skia/effects/SkDashPathEffect.h>
 WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_END
 #include <wtf/CompletionHandler.h>
+#include <wtf/HashMap.h>
+
+class SkSurface;
 
 namespace WebCore {
 
 class WEBCORE_EXPORT GraphicsContextSkia final : public GraphicsContext {
 public:
-    GraphicsContextSkia(SkCanvas&, RenderingMode, RenderingPurpose, CompletionHandler<void()>&& = nullptr);
+    enum class ContextType {
+        RegularContext,
+        RecordingContext
+    };
+
+    GraphicsContextSkia(SkCanvas&, RenderingMode, RenderingPurpose, CompletionHandler<void()>&& = nullptr, ContextType = ContextType::RegularContext);
     virtual ~GraphicsContextSkia();
 
     bool hasPlatformContext() const final;
     SkCanvas* platformContext() const final;
 
     const DestinationColorSpace& colorSpace() const final;
 
+    HashMap<const SkImage*, std::unique_ptr<GLFence>>&& releaseImageFences();
```

Comment:
So this doesn't actually release the fences, it just gives the opportunity for the caller to steal the fences.
So you need to have the signature as 
`HashMap<const SkImage*, std::unique_ptr<GLFence>> releaseImageFences();`

The implementation is fine (return move or return exchange).

Reply:
Good catch, will adapt.